src/pages/LogOutPreviousUserPage.js

```diff
@@ -24,6 +26,10 @@ class LogOutPreviousUserPage extends Component {
                 const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(transitionURL, sessionEmail);
                 if (isLoggingInAsNewUser) {
                     Session.signOutAndRedirectToSignIn();
+                } else {
+                    const exitTo = lodashGet(this.props, 'route.params.exitTo');
+                    Navigation.dismissModal();
+                    Navigation.navigate(exitTo || ROUTES.HOME);
```

Comment:
I think this is creating the problem with the "not found" page. We should only navigate to the workspace after it was created from here https://github.com/Expensify/App/blob/d3120758546153d25002866f6cd770d6b2b123c2/src/libs/actions/Policy.js#L942

Reply:
So we have 2 cases where we're either transitioning to create a new workspace or transitioning to open an existing workspace. This navigation would be used for an existing workspace (as well as any other `exitTo` destination we want to use).

Reply:
Navigation to the exit to route when not logging in a new user is handled here, which might not have been triggered previously with the `componentShouldUpdate` implementation.
https://github.com/Expensify/App/blob/d3120758546153d25002866f6cd770d6b2b123c2/src/libs/actions/App.js#L183-L187

Reply:
Oh okay thanks it looks like this is running but not working because the NavigationContainer isn't ready in time. We might need to add a way to queue navigation for when the container is ready. 

However, this transition logic being split up between AuthScreens calling `App.setUpPoliciesAndNavigate()` and LogOutPreviousUserPage is very confusing. What do you think about consolidating it in LogOutPreviousUserPage and then renaming it to something like TransitionUserPage?

Reply:
> Oh okay thanks it looks like this is running but not working because the NavigationContainer isn't ready in time. We might need to add a way to queue navigation for when the container is ready.

Good discovery that the NavigationContainer isn't ready, I remember that being a problem before. I would undo this commit where you removed that callback to wait until the navigation is ready https://github.com/Expensify/App/commit/e891e7143a98702a818d63d161be511f39c8219c.

> However, this transition logic being split up between AuthScreens calling `App.setUpPoliciesAndNavigate()` and LogOutPreviousUserPage is very confusing. What do you think about consolidating it in LogOutPreviousUserPage and then renaming it to something like TransitionUserPage?

Fun fact, all of the logic for transitions used to be in one component called `LoginWithShortLivedTokenPage`. We separated them into different pages and one action in this PR https://github.com/Expensify/App/pull/8855 and it makes it a lot easier to reason about, because you know roughly what state you are in and each piece of the code only has one concern. I'm open to other ideas about how to simplify it of course. These transitions are a messy beast 👹 😂

Reply:
I think having the un-authenticated vs authenticated components (`LoginWithShortLivedTokenPage` and `LogOutPreviousUserPage`) is fine however it feels like `App.setUpPoliciesAndNavigate()` is basically a big `if` statement and the `else` statement is just nested down in `LogOutPreviousUserPage` for some reason. It's especially awkward because `AuthScreens` and `LogOutPreviousUserPage` are mounted at the same time even when we don't want to log the user out.

Reply:
I'll draft what a combination of `App.setUpPoliciesAndNavigate()` and `LogOutPreviousUserPage` could look like. I think it'll be a bit easier to follow and maintain for our authenticated transitions

Reply:
> it feels like `App.setUpPoliciesAndNavigate()` is basically a big `if` statement and the `else` statement is just nested down in `LogOutPreviousUserPage` for some reason.

Kind of, but only when we need to log out a previous user. Sometimes we just log in the transitioning user. I like `App.setUpPoliciesAndNavigate()` because it has once specific focus.



> It's especially awkward because `AuthScreens` and `LogOutPreviousUserPage` are mounted at the same time even when we don't want to log the user out.

`LoginWithShortLivedTokenPage` is also mounted even when the user is already signed in, but for both cases there's no problem with the page mounting and doing nothing. It will be unmounted when we navigate to the exit route.

I'm quite happy with the current structure of this flow, so if we are going to change it I really want to understand why it's needed. I think there was some talk of making this whole flow into an API command which could be a great way to clean it up. cc @marcaaron because I think you brought this up at one point.

Reply:
Full disclosure I skimmed this thread (sorry I am pre-coffee) but does sound like we could be moving backwards and should maybe create a summary of what the problem is and go from there.

Reply:
I think [this issue](https://github.com/Expensify/Expensify/issues/227553) has the most context from what I can tell. Maybe one of you can post a summary / plan there and we can get some more thoughts.

Reply:
Since fixing these transitions is fairly high priority, I'll update this PR to do that with our current approach and then I'll make a follow up GH to investigate maybe improving the way we handle transitions.